fdbserver/ptxn/TLogPeekCursor.actor.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * TLogPeekCursor.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/ptxn/TLogPeekCursor.h"
+
+#include "fdbserver/ptxn/TLogInterface.h"
+#include "flow/Error.h"
+
+#include "flow/actorcompiler.h" // This must be the last #include
+
+namespace ptxn {
+
+namespace {
+
+// The deserializer will always expect the serialized data has a header. This function provides header-only serialized
+// data for the consumption of the deserializer.
+const Standalone<StringRef>& emptyCursorHeader() {
+	static Standalone<StringRef> empty;
+	if (empty.size() == 0) {
+		TeamID teamID;
+		ptxn::TLogStorageServerMessageSerializer serializer(teamID);
+		serializer.completeMessageWriting();
+		empty = serializer.getSerialized();
+	}
+	return empty;
+};
+
+struct PeekRemoteContext {
+	const Optional<UID> debugID;
+	const TeamID teamID;
+
+	// The last version being processed, the peek will request lastVersion + 1
+	Version* pLastVersion;
+
+	// THe interface to the remote TLog server
```

Comment:
It seems here each team only has one TLog interface? I think the number of TLog interface is determined by replication degree, a typical value is 3. So maybe use a vector?

```suggestion
	// The interface to the remote TLog server
```